MacPorts build fails on M2 Mac Mini (2023) when ffmpeg is installed

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [x] I have read the documentation

### Is your issue present in the latest beta/pre-release?

This issue is present in the latest pre-release

### Describe the Bug

`sudo port install sunshine`

Using https://github.com/LizardByte/Sunshine/releases/download/v2024.910.123325/Portfile

### Expected Behavior

build succeeds

### Additional Context

[main.log](https://github.com/user-attachments/files/16949191/main.log)

I have looked for similar issues and seen #2482 and #3052 but they seem to describe different failures.

### Host Operating System

macOS

### Operating System Version

14.6.1

### Architecture

arm

### Sunshine commit or version

8098df6c381fcb8d9e847144069796114b2dd62c

### Package

macOS - Portfile

### GPU Type

Apple Silicon

### GPU Model

Apple M2

### GPU Driver/Mesa Version

n/a

### Capture Method

None

### Config

```Shell
n/a
```


### Apps

_No response_

### Relevant log output

```shell
full main.log attached above

--- snip ---
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:6:
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/cbs_h264.h:25:
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/cbs.h:28:
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/codec_par.h:206:5: error: unknown type name 'AVChannelLayout'
:info:build     AVChannelLayout ch_layout;
:info:build     ^
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:9:
:info:build /opt/local/include/libavcodec/avcodec.h:227:6: error: redefinition of 'AVDiscard'
:info:build enum AVDiscard{
:info:build      ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/defs.h:210:6: note: previous definition is here
:info:build enum AVDiscard{
:info:build      ^
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:9:
:info:build /opt/local/include/libavcodec/avcodec.h:239:6: error: redefinition of 'AVAudioServiceType'
:info:build enum AVAudioServiceType {
:info:build      ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/defs.h:222:6: note: previous definition is here
:info:build enum AVAudioServiceType {
:info:build      ^
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:9:
:info:build /opt/local/include/libavcodec/avcodec.h:424:16: error: redefinition of 'AVPanScan'
:info:build typedef struct AVPanScan {
:info:build                ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/defs.h:240:16: note: previous definition is here
:info:build typedef struct AVPanScan {
:info:build                ^
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:9:
:info:build /opt/local/include/libavcodec/avcodec.h:453:16: error: redefinition of 'AVCPBProperties'
:info:build typedef struct AVCPBProperties {
:info:build                ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/defs.h:269:16: note: previous definition is here
:info:build typedef struct AVCPBProperties {
:info:build                ^
:info:build In file included from /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/cbs.cpp:9:
:info:build /opt/local/include/libavcodec/avcodec.h:503:16: error: redefinition of 'AVProducerReferenceTime'
:info:build typedef struct AVProducerReferenceTime {
:info:build                ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/third-party/build-deps/ffmpeg/macos-aarch64/include/cbs/defs.h:318:16: note: previous definition is here
:info:build typedef struct AVProducerReferenceTime {
:info:build                ^
--- snip ---
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/upnp.cpp:112:21: error: no matching function for call to 'UPNP_GetValidIGD'
:info:build       auto status = UPNP_GetValidIGD(device.get(), &urls.el, &data, lan_addr.data(), lan_addr.size());
:info:build                     ^~~~~~~~~~~~~~~~
:info:build /opt/local/include/miniupnpc/miniupnpc.h:122:1: note: candidate function not viable: requires 7 arguments, but 5 were provided
:info:build UPNP_GetValidIGD(struct UPNPDev * devlist,
:info:build ^
:info:build /opt/local/var/macports/build/_Users_joe_.dotfiles_macmini_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.23.1/src/upnp.cpp:334:23: error: no matching function for call to 'UPNP_GetValidIGD'
:info:build         auto status = UPNP_GetValidIGD(device.get(), &urls.el, &data, lan_addr.data(), lan_addr.size());
:info:build                       ^~~~~~~~~~~~~~~~
:info:build /opt/local/include/miniupnpc/miniupnpc.h:122:1: note: candidate function not viable: requires 7 arguments, but 5 were provided
:info:build UPNP_GetValidIGD(struct UPNPDev * devlist,
:info:build ^
:info:build 2 errors generated.
--- snip ---
```

--- a/cmake/compile_definitions/common.cmake
+++ b/cmake/compile_definitions/common.cmake
@@ -48,12 +48,12 @@ elseif(UNIX)
     endif()
 endif()
 
-include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third-party/nv-codec-headers/include")
+include_directories(BEFORE SYSTEM "${CMAKE_SOURCE_DIR}/third-party/nv-codec-headers/include")
 file(GLOB NVENC_SOURCES CONFIGURE_DEPENDS "src/nvenc/*.cpp" "src/nvenc/*.h")
 list(APPEND PLATFORM_TARGET_FILES ${NVENC_SOURCES})
 
 configure_file("${CMAKE_SOURCE_DIR}/src/version.h.in" version.h @ONLY)
-include_directories("${CMAKE_CURRENT_BINARY_DIR}")  # required for importing version.h
+include_directories(BEFORE "${CMAKE_CURRENT_BINARY_DIR}")  # required for importing version.h
 
 set(SUNSHINE_TARGET_FILES
         "${CMAKE_SOURCE_DIR}/third-party/moonlight-common-c/src/Input.h"
@@ -129,9 +129,10 @@ list(APPEND SUNSHINE_DEFINITIONS SUNSHINE_PUBLISHER_NAME="${SUNSHINE_PUBLISHER_N
 list(APPEND SUNSHINE_DEFINITIONS SUNSHINE_PUBLISHER_WEBSITE="${SUNSHINE_PUBLISHER_WEBSITE}")
 list(APPEND SUNSHINE_DEFINITIONS SUNSHINE_PUBLISHER_ISSUE_URL="${SUNSHINE_PUBLISHER_ISSUE_URL}")
 
-include_directories("${CMAKE_SOURCE_DIR}")
+include_directories(BEFORE "${CMAKE_SOURCE_DIR}")
 
 include_directories(
+        BEFORE
         SYSTEM
         "${CMAKE_SOURCE_DIR}/third-party"
         "${CMAKE_SOURCE_DIR}/third-party/moonlight-common-c/enet/include"

